streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -1304,22 +1307,72 @@ public void shouldDisableMetricCollectionForAllInternalClients() {
     }
 
     @Test
-    public void shouldDisableMetricCollectionOnMainConsumerOnly() {
+    public void shouldThrowConfigExceptionWhenMainConsumerMetricsDisabledStreamsMetricsPushEnabled() {
+        props.put(StreamsConfig.mainConsumerPrefix(ConsumerConfig.ENABLE_METRICS_PUSH_CONFIG), false);
+
+        final Exception exception = assertThrows(ConfigException.class, () -> new StreamsConfig(props));
+
+        assertThat(
+                exception.getMessage(),
+                containsString("KafkaStreams has metrics push enabled" +
+                        " but the main consumer metrics push is disabled. Enable " +
+                        " metrics push for the main consumer")
+        );
+    }
+
+    @Test
+    public void shouldThrowConfigExceptionConsumerMetricsDisabledStreamsMetricsPushEnabled() {
+        props.put(StreamsConfig.consumerPrefix(ConsumerConfig.ENABLE_METRICS_PUSH_CONFIG), false);
         props.put(StreamsConfig.mainConsumerPrefix(ConsumerConfig.ENABLE_METRICS_PUSH_CONFIG), false);
```

Comment:
Given that we test `consumer.xxx` it seem we should remove this line?

Reply:
I thought we should have a test where `consumerPrefix` metrics is set to `true`,  but users get confused and turn off `mainConsumerPrefix` metrics